File: Linked list/insertion.py
class Node:
  def __init__(self, data):
    self.data = data
    self.next = None


def insert_at_beginning(head, val):
  new_node = Node(val)
  new_node.next = head
  return new_node
  
head = None
head = insert_at_beginning(head, 30)
head = insert_at_beginning(head, 20)
head = insert_at_beginning(head, 10)

#Print the linked list

Curr = head
while Curr:
  print(Curr.data, end=" -> ")
  Curr = Curr.next
print("None")  # To indicate the end of the linked list

File: Linked list/insertion.cpp
#include <iostream>
using namespace std;

struct Node { // Node structure definition
  int data;// Each node contains an integer data and a pointer to the next node

  Node* next; // Pointer to the next node in the linked list
  
  Node(int val){// Constructor to initialize the node with a value. It allows you to create a node with a value directly
    data = val;
    next = nullptr; //NULL pointer initialization
  }
};

//Inserting new node at the beginning of the linked list

void insertAtBeginning(Node*& head, int val) {// Reference to the head pointer and the value to be inserted. & is used so changes to head inside this function are reflected outside too.

  Node* newNode = new Node(val); // Create a new node with the given value

  newNode->next = head; // Point the new node's next to the current head
  
  head = newNode; // Update the head to point to the new node
}